django/contrib/postgres/signals.py

```diff
@@ -1,14 +1,43 @@
+import functools
+
 import psycopg2
 from psycopg2 import ProgrammingError
 from psycopg2.extras import register_hstore
 
+from django.db import connections
+
+
+@functools.lru_cache()
+def get_hstore_oids(connection_alias):
+    with connections[connection_alias].cursor() as cursor:
+        # Retrieve oids of hstore and hstore array.
+        cursor.execute("SELECT t.oid, typarray "
+                       "FROM pg_type t "
+                       "JOIN pg_namespace ns ON typnamespace = ns.oid "
+                       "WHERE typname = 'hstore'")
```

Comment:
I don't have a strong opinion about the quote style but use hanging indent as described in [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).